src/EditorFeatures/CSharpTest/Completion/CompletionProviders/Snippets/CSharpClassSnippetCompletionProviderTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders.Snippets
+{
+    public class CSharpClassSnippetCompletionProviderTests : AbstractCSharpSnippetCompletionProviderTests
+    {
+        protected override string ItemToCommit => "class";
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertClassSnippetInNamespaceTest()
+        {
+            var markupBeforeCommit =
+@"namespace Namespace
+{
+    $$
+}";
+
+            var expectedCodeAfterCommit =
+@"namespace Namespace
+{
+    class MyClass
+    {$$
+    }
```

Comment:
I would like this snippet to work the same as its non-semantic analog:
```
class {|Rename:MyClass|}
{
    $0
}
```

Reply:
Yep you do get the opportunity to rename the MyClass, these tests unfortunately don't reflect that. The location of the last tab-stop is something I'm working to fix for all the snippets I have out now.

Reply:
#63519 

PR to fix cursor location